postgrid/__init__.py

```diff
@@ -470,6 +479,156 @@ def construct_event(cls, payload, secret):
             raise ValueError()
 
 
+def _av_post_request(base, endpoint, json=None, data=None, params=None):
+    data = _map_keys_recursive(data, _snake_to_camel)
+    params = _map_keys_recursive(params, _snake_to_camel)
+
+    headers = {'x-api-key': av_key}
+    data_to_pass = []
+    params_to_pass = []
+
+    def flatten(arr, key, value):
+        if isinstance(value, dict):
+            for inner_key, inner_value in value.items():
+                flatten(arr, f'{key}[{inner_key}]', inner_value)
+        elif isinstance(value, list):
+            for inner_value in value:
+                flatten(arr, f'{key}[]', inner_value)
+        elif isinstance(value, bool):
+            arr.append((key, 'true' if value else 'false'))
+        elif value is not None:
+            arr.append((key, value))
+
+    if data:
+        for key, value in data.items():
+            flatten(data_to_pass, key, value)
+
+    if params:
+        for key, value in params.items():
+            flatten(params_to_pass, key, value)
+
+    res = requests.post(
+        base + endpoint,
+        json=json,
+        data=data_to_pass,
+        params=params_to_pass,
+        headers=headers,
+    )
+    value = res.json()
+
+    try:
+        res.raise_for_status()
+    except requests.HTTPError as e:
+        raise AVError(status_code=res.status_code, message=value['message'])
+    return _av_convert_json_value(value)
+
+
+class RetrieveableAVResource:
+    @classmethod
+    def retrieve(cls, locals_):
+        assert 'cls' in locals_
+
+        locals_except_cls = {
+            key: value for key, value in locals_.items() if key != 'cls'
+        }
+
+        return _av_get(cls.endpoint, **locals_except_cls)
+
+
+class PostableAVResource:
+    @classmethod
+    def post(cls, **kwargs):
+
+        data = kwargs.get('data', None)
+        params = kwargs.get('params', None)
+        json = kwargs.get('json', None)
+
+        return _av_post_request(
+            av_base, cls.endpoint, json=json, data=data, params=params
+        )
+
+
+def _av_get(endpoint, **kwargs):
+    return _request(av_base, endpoint, method='GET', **kwargs)
+
+
+def create_dict(**kwargs):
+    return locals()['kwargs']
```

Comment:
This is effectively just `return kwargs`, so this function isn't necessary

Reply:
Fixed.